src/browser/ui/ReactDOMComponent.js

```diff
@@ -151,6 +154,18 @@ function validateDangerousTag(tag) {
   }
 }
 
+function validateDangerousAttributeName(attributeName) {
+  if (!hasOwnProperty.call(validatedAttributeNameCache, attributeName)) {
+    invariant(
+      VALID_ATTRIBUTENAME_REGEX.test(attributeName),
+      'Invalid attribute name: `%s`',
+      attributeName
+    );
+    validatedAttributeNameCache[attributeName] = true;
+  }
+  return attributeName;
```

Comment:
It's a style that allows you to add validation while passing a value through, similar to function chaining (where a builder returns its self).  Otherwise, the function is just void, so we don't loose anything by returning the value.  This pattern is pretty common in other languages, but we can remove it if it isn't common javascript.